packages/build/src/log/messages/config.js

```diff
@@ -2,15 +2,32 @@
 
 const { cleanupConfig } = require('@netlify/config')
 
+const { DEFAULT_FEATURE_FLAGS } = require('../../core/feature_flags')
 const { omit } = require('../../utils/omit')
 const { logMessage, logObject, logSubHeader } = require('../logger')
 const { THEME } = require('../theme')
 
 const logFlags = function (logs, flags, { debug }) {
+  const flagsA = cleanFeatureFlags(flags)
   const hiddenFlags = debug ? HIDDEN_DEBUG_FLAGS : HIDDEN_FLAGS
-  const flagsA = omit(flags, hiddenFlags)
+  const flagsB = omit(flagsA, hiddenFlags)
   logSubHeader(logs, 'Flags')
-  logObject(logs, flagsA)
+  logObject(logs, flagsB)
+}
+
+// We only show feature flags related to `@netlify/build`.
+// Also, we only print enabled feature flags.
+const cleanFeatureFlags = function ({ featureFlags, ...flags }) {
+  const cleanedFeatureFlags = Object.entries(featureFlags).filter(shouldPrintFeatureFlag).map(getFeatureFlagName)
+  return cleanedFeatureFlags.length === 0 ? flags : { ...flags, featureFlags: cleanedFeatureFlags }
+}
+
+const shouldPrintFeatureFlag = function ([featureFlagName, enabled]) {
+  return enabled && DEFAULT_FEATURE_FLAGS[featureFlagName] !== undefined
+}
+
+const getFeatureFlagName = function ([featureFlagName]) {
```

Comment:
Do you feel it's necessary to have this function, considering that it simply returns the first element of an array? How do you feel about doing the array destructuring as part of the `map()` above?

Reply:
This is a micro-optimization to avoid re-creating the function on each call: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/consistent-function-scoping.md#move-function-definitions-to-the-highest-possible-scope

I think that pattern can also help with re-using the function.

However, an anonymous function works too if you'd prefer :+1:

Reply:
Interesting! I'm all for breaking things out into separate functions once they start getting non-trivial, but I can see things getting a bit hard to manage if we do that on such a micro level.

For example, it's reasonable to imagine another function in the same file also needing to extract the first element of an array, but for a completely different purpose (e.g. `getNameFromPlugin` or something like that). In that case, would you have two methods doing the same thing? Or would you merge them into one? If the latter, how would you name the resulting method? 🤔 

(This is more of an abstract discussion, please feel free to merge the PR as is!)

Reply:
Good points :+1: 
I fixed it :)